### ISSUE TITLE
Configurable client requests timeout

### ISSUE THREAD

**Issue by carlosperate** [MarshalX/atproto#501]
Similar to this issue, but not quite the same, as it's in a different part of the project:
- https://github.com/MarshalX/atproto/issues/489

Essentially when doing a large number of async request I'm seeing a lot of timeouts and increasing the httpx default timeout (via the [httpx timeout constructor parameter](https://www.python-httpx.org/advanced/timeouts/)) resolves it:
https://github.com/MarshalX/atproto/blob/d85a59ffda63a38b01802a4b2c59a82ccdfc41d7/packages/atproto_client/request.py#L173

Would it be possible to provide a user-configurable `timeout` parameter `atproto.Client`/`AsynClient` as well, that is passed to `httpx`?

**Comment by MarshalX** [comment]
Idk is the best way to do so. There are plenty of configuration options. At least three were asked already: retry policy, proxy, and timeout. And for now, the best way to configure it is to make a simple inherit from the class. Like this: https://github.com/MarshalX/atproto/discussions/467#discussioncomment-11384769

**Comment by nlurker** [comment]
Are you talking about InvokeTimeoutError?  I get that a lot.

**Comment by carlosperate** [comment]
> Are you talking about InvokeTimeoutError? I get that a lot.

Yes, this is the exception that bubbles up when `httpx` times out.



> Idk it is the best way to do so. There are plenty of configuration options. At least three were asked already: retry policy, proxy, and timeout. And for now, the best way to configure it is to make a simple inherit from the class. Like this: [#467 (comment)](https://github.com/MarshalX/atproto/discussions/467#discussioncomment-11384769)

That sounds good, as long as it is easy to inherit from `AsyncRequest`, for example something like this:

```python
import asyncio
import httpx
from atproto import AsyncClient
from atproto_client.request import AsyncRequest

class CustomAsyncRequest(AsyncRequest):
    def __init__(self) -> None:
        super().__init__()
        # Here just customise whatever the user needs
        self._client = httpx.AsyncClient(follow_redirects=True, timeout=10.0)

async def foo():
    client = AsyncClient(request=CustomAsyncRequest())
    ...

asyncio.run(foo())
```

Alternatively, if all the configuration needed was in the httpx client, we could just pass one as a `AsyncRequest` constructor argument:

```python
import asyncio
import httpx
from atproto import AsyncClient


async def foo():
    custom_client = httpx.AsyncClient(follow_redirects=True, timeout=10.0)
    client = AsyncClient(client=custom_client)
    ...

asyncio.run(foo())
```

**Comment by MarshalX** [comment]
Configuration has been simplified. More info: https://atproto.blue/en/latest/atproto_client/timeouts.html
